### PR TITLE
Add timestamps to each message

### LINKED ISSUE #9: Add support for timestamps
It could be very useful sometimes to get the timestamps (`-v time`).

### PR DESCRIPTION
**Pull request by vickychijwani**
Enabled with the `--timestamp` flag. Fixes #9.

All I've done is rebase @marczych's commit in PR #34 to the latest master. I've also removed the short `-t` flag because it's already being used for tags. Let me know if any other changes are required to get this merged in.

### CODE REVIEW

File: pidcat.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
 parser.add_argument('-c', '--clear', dest='clear_logcat', action='store_true', help='Clear the entire log before running.')
 parser.add_argument('-t', '--tag', dest='tag', action='append', help='Filter output by specified tag(s)')
 parser.add_argument('-i', '--ignore-tag', dest='ignored_tag', action='append', help='Filter output by ignoring specified tag(s)')
+parser.add_argument('--timestamp', dest='add_timestamp', action='store_true', help='Prepend each line of output with the current time.')
```
Comment (marczych) [inline pidcat.py]: What happened to the first argument (`'-t'`)?

**Comment by marczych** [comment]
Looks good to me other than [my one comment](https://github.com/JakeWharton/pidcat/pull/82/files#r23548856)!

File: pidcat.py (hunk 1 above)
Comment (vickychijwani) [inline pidcat.py]: Removed it because it's already being used to filter by tag.

**Comment by JakeWharton** [comment]
The output looks to be the same. My biggest gripe with the original PR was how it looked. For instance, the date is not useful information.

**Comment by vickychijwani** [comment]
@JakeWharton [I've removed the date already](https://github.com/vickychijwani/pidcat/commit/a897e39685ee7c853550eb9e02df47932d9a23e8). Is there anything else you'd like me to change?

File: pidcat.py [hunk 2]
```diff
@@ -145,7 +146,10 @@ def allocate_color(tag):
 PID_KILL  = re.compile(r'^Killing (\d+):([a-zA-Z0-9._:]+)/[^:]+: (.*)$')
 PID_LEAVE = re.compile(r'^No longer want ([a-zA-Z0-9._:]+) \(pid (\d+)\): .*$')
 PID_DEATH = re.compile(r'^Process ([a-zA-Z0-9._:]+) \(pid (\d+)\) has died.?$')
-LOG_LINE  = re.compile(r'^([A-Z])/(.+?)\( *(\d+)\): (.*?)$')
+if args.add_timestamp:
```
Comment (JakeWharton) [inline pidcat.py]: It doesn't make sense to have this logic sprinkled all over. Always capture the timestamp and just have the condition on whether or not it gets displayed.

Comment (vickychijwani) [inline pidcat.py]: Done.

**Comment by JakeWharton** [comment]
I don't like this output. While you fixed my issues with the timestamp itself simply prepending it to the message is ugly as demonstrated in the original PR.

**Comment by BoD** [comment]
What needs to be done for this to be accepted?  I really like pidcat, but the absence of a timestamp is very frustrating.

**Comment by vickychijwani** [comment]
@BoD I think Jake wants the timestamp to be formatted in a separate column and made visually distinct (like the tags and log levels). I just haven't had time to get to it.

**Comment by jude07** [comment]
@JakeWharton @vickychijwani @marczych @BoD  Has something changed here?? 
I've got pidcat installed and it works fine.But I am having trouble viewing my logs with the timestamps.
I am doing a **pidcat "my-package-name" -t **  , (I've also tried --t , --timestamp)
Is this the correct syntax?

**Comment by marczych** [comment]
@jude07 None of the timestamp pulls have been merged as far as I know so you'll have to use one of the forks to get that option.

**Pull request closed without merging**